ddpui/core/git_manager.py

```
@@ -0,0 +1,576 @@
+
+        :param file_path: Path to the file relative to the git repo root
+        :return: True if file has any changes, False if clean, None if cannot determine
+        """
+        if not self.is_git_initialized():
+            return None
+
+        try:
+            result = self._run_command(["git", "status", "--porcelain", file_path], check=False)
+            return len(result.stdout.strip()) > 0
+
+        except Exception as e:
```

Ruff (0.14.11)

Do not catch blind exception: `Exception`

(BLE001)